framework/db/Connection.php

```diff
@@ -843,6 +843,27 @@ function ($matches) {
     }
 
     /**
+     * Extracts information from [[dsn]]
+     * @param string $name the option name
+     * @return string the required value
+     */
+    public function getDsnInfo($name)
+    {
+        if (preg_match('/\w+:(.*)/', $this->dsn, $matches)) {
+            $options = explode(';', $matches[1]);
+
+            foreach ($options as $opt) {
+                list($key, $value) = explode('=', $opt);
```

Comment:
This is an error for `PHP5`. See [warning](http://php.net/manual/en/function.list.php).